mongodb/accounts.go

```diff
@@ -99,12 +107,16 @@ func NewAccounts(ctx context.Context, db *mongo.Database) (*Accounts, error) {
 	_, err := a.col.Indexes().CreateMany(ctx, []mongo.IndexModel{
 		{
 			Keys: bson.D{{"username", 1}},
-			Options: options.Index().SetUnique(true).
-				SetCollation(&options.Collation{Locale: "en", Strength: 2}),
+			Options: options.Index().
+				SetUnique(true).
+				SetCollation(&options.Collation{Locale: "en", Strength: 2}).
+				SetSparse(true),
```

Comment:
Q: Does the existing index get this new `SetSparse(true)` attribute? Or only gets considered if the index is created for the first time?

Reply:
Unfortunately, tt will have to be part of the migration. Running this against the existing DB will result in:
```
(IndexOptionsConflict) Index with name: username_1 already exists with different options
```
So, we'll have to drop and recreate the index.

Reply:
I'll start on that work now.

Reply:
Cool, sound reasonable. Actually, just asking out of curiosity!